src/Faker/Container/Container.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Faker\Container;
+
+use Psr\Container\ContainerInterface;
+
+/**
+ * A simple implementation of a container.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @experimental This class is experimental and does not fall under our BC promise
+ */
+final class Container implements ContainerInterface
+{
+    /**
+     * @var array
+     */
+    private $container = [];
+    private $cache = [];
+
+    public function __construct(array $container)
+    {
+        $this->container = $container;
+    }
+
+    public function get($id)
+    {
+        if (isset($this->cache[$id])) {
+            return $this->cache[$id];
+        }
+
+        if (!$this->has($id)) {
+            throw new NotInContainerException(sprintf('There is not service with id "%s" in the container.', $id));
+        }
+
+        $data = $this->container[$id];
+        if (is_string($data)) {
```

Comment:
```suggestion
        if (is_string($data) && class_exists($data)) {
```

Reply:
I was considering this too. But with this change the exception is all wrong when you trying to get a service that does not exist.

Reply:
I've updated the code to check if the class exists.